### ISSUE TITLE
virtual-modules section: docs

### ISSUE THREAD

**Issue by ciez** [haskell/cabal#7602]
Could anyone hint / share links to a description of .cabal file virtual-modules section?

The docs show but a placeholder:
https://cabal.readthedocs.io/en/3.4/cabal-package.html?highlight=virtual-modules#pkg-field-virtual-modules

**Comment by Mikolaj** [comment]
That's just a flat list of modules, just as exposed-modules.

Some snippets from `git grep -30 virtual-modules`:

Cabal/ChangeLog.md:  * Added `virtual-modules` field, to allow modules that are not built
Cabal/ChangeLog.md-    but registered (#4875).

doc/cabal-package.rst:.. pkg-field:: virtual-modules: identifier list
doc/cabal-package.rst-    :since: 2.2
doc/cabal-package.rst-
doc/cabal-package.rst-    A list of virtual modules provided by this package.  Virtual modules
doc/cabal-package.rst-    are modules without a source file.  See for example the ``GHC.Prim``
doc/cabal-package.rst-    module from the ``ghc-prim`` package.  Modules listed here will not be
doc/cabal-package.rst-    built, but still end up in the list of ``exposed-modules`` in the
doc/cabal-package.rst-    installed package info when the package is registered in the package
doc/cabal-package.rst-    database.

**Comment by phadej** [comment]
TL;DR you don't need `virtual-modules` (except if you are GHC developer). It's a feature solely for GHC need: [GHC.Prim module in `ghc-prim`](https://hackage.haskell.org/package/ghc-prim-0.6.1/docs/GHC-Prim.html) doesn't really exist as a source. GHC just knows it.

**Comment by ciez** [comment]
Cheers Mikolaj, Oleg.

So for all practical purposes could this file be added in ghc-prim/GHC folder to the same effect:
[Prim.hs](https://github.com/haskell/cabal/files/7082730/Prim.hs.txt)
?

**Comment by phadej** [comment]
@ciez no. That would be a different module. It also may confuse build system, if source existed. `GHC.Prim` contains the primops, which are hard-wired into compiler. I think things (like costom name resolver) just have to know about that corner case, i.e. that `virtual-modules` (which is only one) doesn't exist. E.g. the module finder in GHC itself has following code:

```haskell
   -- special case for GHC.Prim; we won't find it in the filesystem.
   -- This is important only when compiling the base package (where GHC.Prim
   -- is a home module).
   if mod `installedModuleEq` gHC_PRIM
         then return (InstalledFound (error "GHC.Prim ModLocation") mod)
         else searchPathExts home_path mod exts
```

**Comment by ciez** [comment]
So in this particular case, what symbols does GHC.Prim export?

All symbols exported by 
  module GHC.CString,
  module GHC.Classes,
  module GHC.Debug,
  module GHC.IntWord64,
  module GHC.Magic,
  module GHC.Prim.Ext,
  module GHC.Tuple,
  module GHC.Types
 or ..?

**Comment by phadej** [comment]
[The haddocks](https://hackage.haskell.org/package/ghc-prim-0.6.1/docs/GHC-Prim.html) are correct. The primitive types like `Int#` and operations on them like `(+#)`

**Comment by phadej** [comment]
I found this: https://downloads.haskell.org/ghc/9.2.1-rc1/docs/html/libraries/ghc-prim-0.8.0/src/GHC.Prim.html

**Comment by ciez** [comment]
Cheers Oleg! 
This is just what I was looking for.

Appreciate this.
